Remove malc0de feed due to offline

gutsohnCERT opened the pull request:
Remove malc0de parser, adapt collector and parser settings in the default configuration, prepare upgrade function warning and edit the entry in feeds.yaml

FIX: #2178

gutsohnCERT committed:
FIX: Remove malc0de parser (The malc0de feed seems to be discontinued)

Fix #2178

gutsohnCERT committed:
FIX: Add v310_deprecations in order to match mapping

Fix #2178

gutsohnCERT committed:
FIX: Add removed malc0de tag to CHANGELOG.md

Fix #2178

sebix commented on CHANGELOG.md:
```diff
@@ -53,6 +53,7 @@ CHANGELOG
   - SMTP Data
   - Telnet Login
   - VNC/RFB Login
+- Removed `intelmq.bots.parsers.malc0de`: this bot was marked as deprecated and removed from feed due to offline status (PR#2184 by Tamas Gutsohn, FIX#2178).
```
```suggestion
- Removed `intelmq.bots.parsers.malc0de`: this bot was marked as deprecated and removed from feed due to offline status (PR#2184 by Tamas Gutsohn, fixes #2178).
```

sebix replied:
```suggestion
- Removed `intelmq.bots.parsers.malc0de`: this bot was marked as deprecated and removed from feed due to offline status (PR#2184 by Tamas Gutsohn, fixes#2178).
```

sebix reviewed (changes requested):
The upgrade function does not discover the collectors and is missing a test function

sebix commented on intelmq/lib/upgrades.py:
```diff
@@ -725,31 +731,49 @@ def v301_deprecations(configuration, harmonization, dry_run, **kwargs):
+
+
 UPGRADES = OrderedDict([
-    ((1, 0, 0, 'dev7'), (v100_dev7_modify_syntax, )),
```
Are these changes necessary (as part of this PR)?

wagner-intevation committed:
doc: minor change for PR#2184 in changelog

wagner-intevation committed:
Merge branch 'develop' into gutsohnCERT_set_malc0de_feed_offline

wagner-intevation committed:
malc0de deprecation: check for collector in upgrade function

the feed malc0de is offline. in the upgrade function, also check for the
collector
